tests/test_new_return_types_qnode.py

```diff
@@ -0,0 +1,1665 @@
+        import tensorflow as tf
+
+        dev = qml.device(device, wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="tf")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.PauliZ(wires=1))
+
+        x = tf.Variable([0.1, 0.2, 0.3])
+
+        with tf.GradientTape() as tape:
+            out = circuit(x)
+
+        res = tape.jacobian(out, x)
+
+        qml.disable_return()
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2, 3)
+
+    @pytest.mark.tf
+    def test_multiple_meas_tf_autograph(self):
+        """Return Jacobian of multiple measurements with Tf Autograph."""
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit", wires=2)
+        qml.enable_return()
+
+        @tf.function
+        @qml.qnode(dev, interface="tf")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.PauliZ(wires=1))
+
+        # Autograph does not support multiple measurements with different shape.
+
+        x = tf.Variable([0.1, 0.2, 0.3])
+
+        with tf.GradientTape() as tape:
+            out = circuit(x)
+
+        res = tape.jacobian(out, x)
+        qml.disable_return()
+
+        assert isinstance(res, tf.Tensor)
+        assert res.shape == (2, 3)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("device", devices)
+    def test_multiple_expval_jax(self, device):
+        """Return Jacobian of multiple expvals."""
+        import jax
+
+        dev = qml.device(device, wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.expval(qml.PauliZ(wires=0)), qml.expval(qml.PauliZ(wires=1))
+
+        x = jax.numpy.array([0.1, 0.2, 0.3])
+        res = jax.jacobian(circuit)(x)
+
+        qml.disable_return()
+
+        assert isinstance(res, tuple)
+        assert len(res) == 2
+        for elem in res:
+            assert isinstance(elem, jax.numpy.ndarray)
+            assert elem.shape == (3,)
+
+    @pytest.mark.autograd
+    @pytest.mark.parametrize("device", devices)
+    def test_multiple_probs_autograd(self, device):
+        """Return Jacobian of multiple probs."""
+        dev = qml.device(device, wires=2)
+        qml.enable_return()
+
+        @qml.qnode(dev)
+        def circuit(a):
+            qml.RX(a[0], wires=0)
+            qml.CNOT(wires=(0, 1))
+            qml.RY(a[1], wires=1)
+            qml.RZ(a[2], wires=1)
+            return qml.probs(op=qml.PauliZ(wires=0)), qml.probs(wires=1)
```

Comment:
What if the second probs acts on 2 qubits? That would have created a ragged array before.

Reply:
Yes correct you can see the `test_multiple_meas_autograd` test for this case, but ragged array boxes raise no warnings.